spid-spring-integration/src/main/java/it/italia/developers/spid/integration/util/SPIDIntegrationUtil.java

```diff
@@ -139,7 +139,8 @@ public String printAuthnRequest(AuthnRequest authnRequest) throws IntegrationSer
 		try {
 			Signer.signObject(authnRequest.getSignature());
 		} catch (SignatureException e) {
-			  e.printStackTrace();
+			log.error("There was an error while signing the request", e);
+			throw new IntegrationServiceException(e);
```

Comment:
Consider avoiding this log-and-throw pattern to avoid log duplication, maybe have a read here https://stackify.com/best-practices-exceptions-java/ (point 8).

Reply:
removed log statement